src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -62,13 +64,52 @@ public static bool TryGetCounterPayload(this TraceEvent traceEvent, CounterFilte
                     displayUnits,
                     value,
                     counterType,
-                    intervalSec);
+                    intervalSec,
+                    metadataDict);
                 return true;
             }
 
             return false;
         }
 
+        public static Dictionary<string, string> GetMetadata(string metadataPayload)
+        {
+            var metadataDict = new Dictionary<string, string>();
```

Comment:
Consider adding some comments or link a github issue to indicate the limitations of the metadata string (which should really be an array, but we can't change that now)